nautobot/extras/filters/__init__.py

```diff
@@ -387,6 +387,11 @@ class NautobotFilterSet(
     are needed.
     """
 
+    def filter_queryset(self, queryset):
+        # Ensure the filtered result is distinct, particularly when filtering on a
+        # many-to-many field with `conjoined` set to False to avoid duplicate entries.
+        return super().filter_queryset(queryset).distinct()
```

Comment:
On re-reading the django-filter docs, it should be possible on an individual filter to specify `distinct=True` rather than this broad-spectrum approach, but furthermore, a `MultipleChoiceFilter` already defaults to `distinct=True`. I think we actually have a bug in `ContentTypeMultipleChoiceFilter.filter()` not taking `self.distinct` into account.